### ISSUE TITLE
MIDI File plugin file selector does not show any files

### ISSUE THREAD

**Issue by SpotlightKid** [falkTX/Carla#618]
- Add a "MIDI File" internal plugin
- Click File icon to open file selector to load a MIDi file

==> File selector does not show any files, only directories.

Selecting the (second) empty option in the file filter drop-down doesn't make any difference.

This makes it impossible to load/change a MIDI file in an existing "MIID File " plugin instance.

![carla-midi-file-plugin-file-select](https://user-images.githubusercontent.com/5719151/38582763-1dd2a51e-3d11-11e8-8198-02c22422d045.png)

**Comment by SpotlightKid** [comment]
Any comments on this?

Can anybody reproduce it?

**Comment by pdesaulniers** [comment]
Yes, but selecting the empty option does make the files show up for me.

I'm also using Arch, so I suspect this is why we're getting strange issues like these.

**Comment by trebmuh** [comment]
As a workaround, you can drag & drop a midi file from the file browser (left side from Carla's main windows) on the plugin.

@falkTX [this](https://github.com/falkTX/Carla/blob/master/source/native-plugins/midi-file.cpp#L97) looks a bit strange-ish to me, specially the  `"MIDI Files *.mid;*.midi;;"` part. Maybe something like  `"MIDI Files *.mid; *.midi;"` would work better? (You know that better than me obviously!)

**Comment by falkTX** [comment]
Adjusted the filter to look exactly like the Qt examples.
Let me know if that fixes it for you, thanks.

**Comment by pdesaulniers** [comment]
Fixed for me.

**Comment by falkTX** [comment]
@SpotlightKid please confirm
